### DISCUSSION TITLE
WARNING: USE A VPN. Genmon exposed to the internet was remotely started

### DISCUSSION THREAD

**Discussion by wotseurba** [jgyates/genmon#1029] [General]
For the past 2 days, my generator has started up (not transferred) for no reason (no outage). The first time I didn't know what was happening, reviewed my Genmon status to see only that the engine was running, with the log stating "remote start" which I didn't do.  It took me a few minutes to assess the situation, and then shut down the generator under the maintenance tab. 

Last night, same thing happened.  Except, my site name has changed to: "Hacked by Nariozon"

So, someone has figured out how to log into my genmon, and is doing this.  What should I do?

I'd like to change the IP port to something other than 8000, so I can close off the port forwarding of 8000.

Has anyone else experienced this?

Logs submitted

### Your Environment

- Generator Model: 20kw Evolution
- Genmon Logs: Submit via the About page
- Genmon Version: V1.18.10

**Comment by skipfire** [comment]
Genmon's port can be changed in /etc/genmon/genmon.conf.  There is a setting for http_port.  If you are wanting to expose genmon to the internet on any port you should probably switch to SSL at a minimum, though you may want to see if your router supports inbound VPN so you can VPN in instead of exposing genmon at all.  And of course, I would suggest changing both the username and password you use for genmon.

**Comment by djrallard** [comment]
I’d recommend a new SD card and rebuild your GenMon install. You can’t know what they might have installed or done to your installation. 
Then, use a strong password, MFA, and change your workflow to use a VPN for access to your internal network instead of exposing devices directly to the internet.

**Comment by wotseurba** [comment]
Thank you both for the comments and recommendations.  I've powered-down the Genmon Pi and closed up the router port forwarding 8000.  I will make an assessment of my router's capability of inbound VPN before moving forward to rebuild of my Genmon.  I find it strangely interesting (and disturbing) that someone actually knows the workings of Genmon, and gains satisfaction my turning on my generator remotely.  And, changing my site name, thus exposing themselves as doing this.  When my generator started up at 3am, with no power outage, and not scheduled exercise, it had me wondering if my Evolution controller had failed, or Genmon has some glitch.  But after it happened again last evening, and the hacker changed my site name, I now know the cause.  I thought it would be a heads-up to other users and jgyates might offer a security alert and advice for users to use VPN, and-or using something other than port 8000.

**Comment by jgyates** [comment]
I agree that this is a good reminder for everyone exposing genmon to the internet to everyone.

Here are all of the times VPN was advised in the issues:

https://github.com/jgyates/genmon/issues?q=is%3Aissue+vpn

And the front page of the wiki also advises against this in the Connectivity section:

https://github.com/jgyates/genmon/wiki#connectivity

From an intruders point of view, changing the port and using SSL has little to no effect to a hacker. The path they are likely taking is probing your router for exposed ports. This is easily done by scanning consecutive IP addresses probing all portss until they find something open. Your exposed port will eventually be hit.  If they find your open port they try to look for a web interface on that port. This brings up the genmon logon page. From there they just google genmon and they will find the github project with 100% of the source code since this is an open source project. They don't even need the source code to use one of hundreds of publicly available password cracker programs. A 7 character (or less) password can be cracked by a computer in less than a second. Once they have your password cracked then what they do is up to them. Best case scenario they let you know you are exposed, like your hacker appeared to do to you. Worst case they try to gain access to your pi (not just he web interface) and then launch attacks on your local network, possibly gaining access to other computers.

In addition, the web server used by genmon is not intended for production use. It is the web server built into the flask libraries and the developers of that software do not recommend using it in a production environment as the web server was not built with security in mind. It would not be resilient to attacks like a denial of service or other non password cracking hacks. 

In short, don't expose genmon to the internet. **If you must** access genmon from outside your network, use a VPN as it is much harder to hack and always use strong passwords. 

I am editing the title of this thread to better warn everyone to use a VPN.

**Comment by jgyates** [comment]
Also, as mentioned by @skipfire, enabling genmon MFA would go a long way to defeating people trying to do a simple password crack.

**Comment by wotseurba** [comment]
Thank you

**Comment by jgyates** [comment]
Two notes on this thread to close it out:

1) There is a new version of genmon today (V1.18.11) that implements a temporary lockout after the maximum number of failed logins after a period of time. This feature has two parameters (max failed login attempts and lockout duration in seconds) both of which are on the Advanced settings page. The defaults are 5 and 300 seconds (5 min). For example with the defaults if there are 5 failed logins within 5 minutes then the login is locked out for 5 minutes. This should deter most brute force password cracking attempts for reasonable lengths passwords.

2) I examined the logs for the hack in question. In this instance the port was exposed but no username and login were used so it looks like someone just did a port scan and found an open port and pointed a browser at it. No password cracking.

**Comment by wotseurba** [comment]
Thank you for the efforts you expended on this.  I sent you a (modest) donation yesterday.

 

I also did a fresh install on a new SD card yesterday, and also updated to V1.18.11.  I've never used a login for Genmon, but might consider it.

 

Also, the new install was extremely easy to accomplish with both the latest OS, and the Genmon installation/configuration script.  I remember doing my first install many years ago, and having to manually configure things like serial port, etc.  Much easier and automatic now.

 

Thanks again

Brian Ruestow

 
 

Sent: Tuesday, April 19, 2022 at 7:12 PM

**Comment by mattm55** [comment]
Tonight I just had my Generac start in the middle of the nite. Same thing, hacked. Settings were changed. Time to lock things down here. Rebuilding PI &  Genmon tomorrow.

**Comment by jgyates** [comment]
Thanks for the info.

**Comment by jgyates** [comment]
I have updated the wiki, referencing this thread and pinned this thread in hopes of getting more visibility.

**Comment by mattm55** [comment]
Great and Thanks. I love "genmon" so it is something I need to completely reinstall, with the proper security. 

I woke up to the gen running at midnight, determined it wasn't my utility and shutdown thru genmon. 10 minutes later it started again. Before I dropped the Pi after shutting down the 2nd time was to see all my settings were changed (excercise time, site name, config (24v over 12v, email, etc...) Grrrr.... can only blame myself.

**Comment by diverbill** [comment]
If you are looking for a VPN solution one i have switched to and really like is the Cloudflare Zero Trust service. https://developers.cloudflare.com/cloudflare-one/ With a little tweaking i have an always on vpn that routes 90% of my DNS connections back to my house for 
A) dns security with pi-hole & local DNS
B) access to all my local service GENMON Security, cameras, NAS Shares
C) no open internet ports for someone to access
The one requirement is you must own a domain

**Comment by lmamakos** [comment]
I've also had great success using tailscale.com as a VPN between your devices.  There's clients for mobile devices, windows, mac and Linux.  I run the tailscale client on a few Raspberry Pi systems, including the one used for genmon.   It's free for a whole bunch of devices, and the paid version is really only required for multi-user enterprise sorts of applications.

**Comment by spatel7786** [comment]
Hello all,

I'm new to all of this so please excuse my ignorance. I'm in the process of getting my genmon all put together and came across this thread about using VPN. Will i be safe if the genmon is connected to wifi/ethernet and only accessed when im home in my network? I have no plans to access this outside of my home. Would i still need to use VPN for my scenario? @jgyates

**Comment by jgyates** [comment]
If you only access genmon on your local network then you don't need a VPN. In this scenario your Internet router is providing protection from attacks. If you want to access genmon, or anything on your local network from outside your network, then a VPN is the way to go to maintain safety.

**Comment by daved44** [comment]
I am getting a header on my genmon dashboard: "Injection by Nariozon". It started today.  Genmon 2.0.0, Ethernet attached. No other details yet, but I simply disconnected the Ethernet cable and unplugged the Pi until I have more information and time to get to it. I have been accessing it remotely and would like to continue to do so but hesitate to use a VPN unless I really have to. I have a backup SD Card and could use it after turning off WAN access in my router (hopefully temporarily). But that backup is over a year old and if I go to it and update to the latest Genmon, I lose all logs etc. since the backup, which document some significant utility problems since then. 

The generator doesn't appear to have been run since yesterday morning's exercise, as it is cool and there are no additional run logs.

Anyone else seeing this? It's been over two years since anyone updated this discussion. Thoughts, please, on insulating genmon from the internet (aside from what has already been said above) and getting my log, failure, notes, etc., to my new genmon SD card.

Screenshot attached. 

<img width="1013" height="672" alt="Screenshot 2026-05-13 063949" src="https://github.com/user-attachments/assets/0b238dfc-5f25-408e-9bb9-a4a4abf6b6be" />

**Reply by skipfire** [reply]
Did you have security enabled on your exposed genmon?  Did you enable MFA?

Why do you hesitate to use VPN? It's the right tool for safely accessing a system remotely, though tailscale would be another option.

You can grab the kwlog.txt from /etc/genmon/ and you can grab any other logs you want from /var/log/, do your restore from backup, then restore those log files.  kwlog.txt is also in the exported configuration backup.

**Reply by skipfire** [reply]
Also that `Injection by Nariozon` is the site name configured in genmon.

**Reply by daved44** [reply]
> Also that `Injection by Nariozon` is the site name configured in genmon.

What exactly do you mean? Has the site name been changed by nariozon with the hack?

I have hesitated to use a VPN across my local network generally, mostly because it is something else that needs be researched for which I don't have the time or ambition and, as with everything, a VPN will likely present me with some unintended consequences. But in my ongoing effort to get away from Gaggle, I moved to proton mail a couple days ago and in the process of checking out proton  I turned on the proton VPN yesterday and as of this morning I have lost access to mapped drives on the server I turned it on for (IMO an unintended consequence) and will have to shoot that bug this morning. 

My several Pi's are backed up using image copies, so they are complete. 

I will be disabling all WAN ports on my router until I solve this (and possibly other) problems.  Thanks for the info on copying my genmon logs etc. from the hacked SD card. 

Too many electronic devices, but I keep adding and I'm getting too old for this. My home is reaching critical mass. Expect an implosion soon...

Please keep the comments coming!

**Reply by skipfire** [reply]
It's not really a hack, he just edited the site name in the settings page in genmon.

Do you have security enabled on genmon and MFA, or is it just wide open?  A simple username and password provides some security and MFA adds even more, though still VPN or tailscale with a login is best so that genmon isn't even exposed to the internet.

If you feel VPN is too much for you, check out tailscale. It's extremely easy to setup and is an additive network. VPNs can be configured to isolate connectivity to the VPN which may be what happened with your VPN setup.

**Reply by djrallard** [reply]
> “I have hesitated to use a VPN across my local network generally, mostly because it is something else that needs be researched for which I don't have the time or ambition and, as with everything, a VPN will likely present me with some unintended consequences. “

I highly recommend checking out Tailscale. It uses WireGuard and is super simple to setup and you don’t have to expose any outside ports on your firewall. Well documented and supported. Much easier to manage than Cloudflare Tunnels.

**Comment by mattm55** [comment]
Yes, backup your logs and files or the whole instance but proceed with caution as any direct copy could have corrupted code.

I had the hacker start my gen at midnight on and off. There are enough ways to secure this, VPN and other. Also, make sure any other open remote connections haven't been hacked. You might not even know? Opening any remote connection is susceptible to hacking.

**Reply by jgyates** [reply]
@daved44 

I assuem you are using a user name and password, correct? I looks like in this case you intruder is letting you know you have a weakness by changing your site name on the settings page. They either guessed your password by using some type of automation or you have some other intrusion (password manager leaked, trojan, etc). 

Do you have email enabled and working on your genmon install? If you do you should get a warning email if genmon is locked due to too many failed password attempts (i.e. the password was guessed after multiple attempts. 

@skipfire is correct. Using Multi-factor authentication (MFA) can help keep out people who attempt to guess your password.

**Comment by daved44** [comment]
Thanks, all. To fix this, it is apparent that I need to turn Genmon back on, Pi and ethernet.  Is it certain enough that nariozon hacked only the username and I may be able to go forward with this on the WAN after implementing additional security and without going to the backup?

**Reply by jgyates** [reply]
Hi @daved44 

The first thing I would do is disable remote access at your browser.  This will prevent anyone outside of your local network from accessing genmon through any exposed ports.

I don't think it is certain that the intruder only accessed your device through username and password. We don't have enough information at this point to say that. This is why I was asking about if you had email enabled. If you did and got notifications of multiple bad passwords then that would be a vote for the intruder. Guessing your password. outside of this information we really don't have any info to tell us how they got into your system. 

Do you have email enabled? If so, you can send me your logs and I can take a look to see if there's something that would indicate a password attempt?

**Comment by mattm55** [comment]
Adding security is not optional today.

**Comment by daved44** [comment]
I'm setting up tailscale; so far so good. When done I'll reconnect the generator and look through genmon, then get back. With time constraints, it may be a while.

BTW, excellent support as usual and 2.0.0 looks really good!

**Comment by daved44** [comment]
I do open the genmon interface daily when I first get up so i know that the problem just occurred last night. I'm still in the process of installing Tailscale; so far so good but have a bunch of devices to install it on. 
Will I need do anything else to secure it (and the rest of my world)? WAN port forwarding still off in my router; I will turn it back on after everything is secure. Unfortunately, this is taking the better part of yet another day when I would rather be doing something productive outside on a beautiful spring day.

Email output is set, but I was not securing Genmon with a password. When originally installed I used it only locally; I have since opened a port to access it when I'm away from home, but didn't secure it then. It is now secured with user name and pw.  I did get two email notifications last night, the first which stated only:
Time: 04:09:44 AM
Date: Wed 13-May-2026
Generator Monitor Starting at Injection by Nariozon using IP address 192.168.50.70

The second starts with the following:
Time: 04:10:00 AM
Date: Wed 13-May-2026
NOTE: This message is a notice that the state of the generator has changed. The system is not in alarm.
Status : 
    Engine : 
        Switch State : Auto
        Engine State : Off - Ready
etc...

**Reply by jgyates** [reply]
As a fyi, there have been at least instances that 3 that I know of where unauthorized access to genmon has occurred. All of these are when genmon was directly exposed on the internet. After the first two we added some additional checks so when people use a brute force attack to guess your password you get locked out for a period of time and an email is sent. 

If you allow internet access to genmon and do not have a username / password, this will happen again.

Also, if you setup tailscale, you do not need port forwarding on your router to be enabled so there should be no need to do that. Tailscale is like a point to point VPN.

The emails you got were when the person saved the change to your site name. This triggers a restart which sends out the two emails described above.

**Comment by daved44** [comment]
Tailscale went well. Best thing since sliced bread and thank you for the suggestion. Port forwarding is turned off permanently and I can get to all of my devices from all of my devices, including shares, even away from home. Only thing left to do is to give access to my weather station to my son in another town, and it looks like Tailscale funneling may be the answer, but I have to do a little learning about it. 

I am having one problem, having to do with logging into Genmon. 

First, it appears that once logged in I don't have to do it again, or that appears to be the case with the desktop server. Perhaps not across a reboot, but I don't reboot the desktop often.

Second, I am unable to login on a W11 laptop. I get the Genmon login screen and enter the username/password, but this information immediately disappears and I can try, try again until the number of attempts are exceeded and I am locked out. This is with Edge, as is the desktop, and even with all ad blockers, trackers, and internal ad blocking turned off. I was able to login with a clean vanilla version of Firefox, which I don't use but keep for these kinds of things. 

Easy fix or should I start and issue?

**Reply by skipfire** [reply]
I have no problem with edge, the behavior you are describing is just bad user name/password flow. You may want to check your config file to ensure you are using the right credentials to try logging in.

**Reply by jgyates** [reply]
I have no problem with my testing with Edge on Windows 11 either. Are you using a password saved by the browser or a password manager? If you are you may want to check if they have been synced across the devices you are using.

**Reply by daved44** [reply]
I had tried by using the password manager, which I determined was up to date (synced), and by entering the userid/password manually. No cigar. 
Then I went into Genmon on the other browser and updated the security page to get a trusted CA, which worked as advertised. Sometime during the time I was messing around with this and perhaps other things, it began working from all locations in my tailnet. 

It would be impossible for me to describe where I was and what I did between the time it didn't work and it did work, so the best thing to do with this is nothing. It works.

**Comment by wtnh** [comment]
I finally got around to installing Tailscale and so far so good. I have been using Genmon for years and I am now able to get into the Genmon dashboard from my iPhone when away. It seems to coexist on the iPhone with Apple Private Relay without issue. I just have to switch it on and then touch a saved Safari link on my phone's home screen pointing to the Tailscale IP address they assigned for my Genmon. The GUI looks great on the phone screen.

While I was at it, I threw caution to the wind and installed log2ram.
